Source/WebCore/Modules/webtransport/WorkerWebTransportSession.cpp

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "WorkerWebTransportSession.h"
+
+#include "WebTransport.h"
+#include "WebTransportBidirectionalStreamConstructionParameters.h"
+#include "WebTransportReceiveStreamSource.h"
+#include "WorkerGlobalScope.h"
+#include "WorkerThread.h"
+#include "WritableStreamSink.h"
+
+namespace WebCore {
+
+Ref<WorkerWebTransportSession> WorkerWebTransportSession::create(WorkerGlobalScope& scope, WebTransportSessionClient& client, Ref<WebTransportSession>&& session)
+{
+    return adoptRef(*new WorkerWebTransportSession(scope, client, WTFMove(session)));
+}
+
+WorkerWebTransportSession::~WorkerWebTransportSession() = default;
+
+WorkerWebTransportSession::WorkerWebTransportSession(WorkerGlobalScope& scope, WebTransportSessionClient& client, Ref<WebTransportSession>&& session)
+    : m_scope(scope)
+    , m_client(client)
+    , m_session(WTFMove(session))
+{
+    m_session->attachClient(*this);
```

Comment:
I don't understand how safer cpp thinks this is uncounted and unsafe.  m_session is a Ref

Reply:
If `m_session` never gets reassigned then mark it as `const`. Otherwise, you need a `Ref` the the session *on the stack* for safety.